tests/Unit/Elliptic/Actions/Test_InitializeAnalyticSolution.cpp

```diff
@@ -0,0 +1,107 @@
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/ElementIndex.hpp"
+#include "Domain/Tags.hpp"
+#include "Elliptic/Actions/InitializeAnalyticSolution.hpp"
+#include "Parallel/ParallelComponentHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/ActionTesting.hpp"
+
+namespace {
+
+struct ScalarFieldTag : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+template <size_t Dim>
+struct AnalyticSolution {
+  tuples::TaggedTuple<ScalarFieldTag> variables(
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& x,
+      tmpl::list<ScalarFieldTag> /*meta*/) const noexcept {
+    Scalar<DataVector> solution{2. * get<0>(x)};
+    for (size_t d = 1; d < Dim; d++) {
+      get(solution) += 2. * x.get(d);
+    }
+    return {std::move(solution)};
+  }
+  // clang-tidy: do not use references
+  void pup(PUP::er& /*p*/) noexcept {}  // NOLINT
+};
+
+template <size_t Dim>
+struct AnalyticSolutionTag : db::SimpleTag {
+  using type = AnalyticSolution<Dim>;
+};
+
+template <size_t Dim, typename Metavariables>
+struct ElementArray {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementIndex<Dim>;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Initialization,
+          tmpl::list<ActionTesting::InitializeDataBox<
+              tmpl::list<::Tags::Coordinates<Dim, Frame::Inertial>>>>>,
+
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Testing,
+          tmpl::list<elliptic::Actions::InitializeAnalyticSolution<
+              AnalyticSolutionTag<Dim>, tmpl::list<ScalarFieldTag>>>>>;
+};
+
+template <size_t Dim>
+struct Metavariables {
+  using element_array = ElementArray<Dim, Metavariables>;
+  using component_list = tmpl::list<element_array>;
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+template <size_t Dim>
+void test_initialize_analytic_solution(
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& inertial_coords,
+    const Scalar<DataVector>& expected_solution) {
+  using metavariables = Metavariables<Dim>;
+  using element_array = typename metavariables::element_array;
+  const ElementId<Dim> element_id{0};
+  ActionTesting::MockRuntimeSystem<metavariables> runner{
+      {AnalyticSolution<Dim>{}}};
+  ActionTesting::emplace_component_and_initialize<element_array>(
+      &runner, element_id, {inertial_coords});
+  runner.set_phase(metavariables::Phase::Testing);
+  ActionTesting::next_action<element_array>(make_not_null(&runner), element_id);
+  const auto get_tag = [&runner, &element_id](auto tag_v) -> decltype(auto) {
```

Comment:
Since the lambda is only used once can you omit it?

Reply:
I like to set up the `get_tag` lambda so that it's easy to add more checks later if needed. Are you sure you prefer to remove it?

Reply:
Yes